File: src/main/java/org/fountainmc/api/world/block/plants/Leaves.java
package org.fountainmc.api.world.block.plants;

public interface Leaves extends Plant {

    /**
     * Return if the leaves can decay
     *
     * @return if able to decay
     */
    boolean canDecay();

    LeafType getType();

    enum LeafType {
        OAK,
        SPRUCE,
        BIRCH,
        JUNGLE,
        ACACIA,
        DARK_OAK;
    }

}

File: src/main/kotlin/org/fountainmc/api/world/block/plants/Leaves.kt
package org.fountainmc.api.world.block.plants

interface Leaves : Plant {

    /**
     * Return if the leaves can decay

     * @return if able to decay
     */
    fun canDecay(): Boolean

    val type: LeafType

    enum class LeafType {
        OAK,
        SPRUCE,
        BIRCH,
        JUNGLE,
        ACACIA,
        DARK_OAK
    }

}
